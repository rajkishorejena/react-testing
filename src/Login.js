import React, { useState } from "react";

const Login = () => {
  const [userName, setUsername] = useState("");
  const [initialState, SetInitialState] = useState("");
  return (
    <div
      style={{
        width: "100%",
        minHeight: "100vh",
        display: "flex",
        justifyItems: "center",
        alignItems: "center",
      }}
    >
      <input
        type="text"
        value={userName}
        onChange={(e) => setUsername(e.target.value)}
        name="username"
        placeholder="please enter your name"
      />
      <label>{initialState}</label>
      <button onClick={() => SetInitialState("Update Button Clicked")}>
        Update
      </button>
    </div>
  );
};

export default Login;
