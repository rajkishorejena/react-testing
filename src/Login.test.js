import { render } from "@testing-library/react";
import Login from "./Login";
describe("Snapshot Testing", () => {
  test("Snapshot testing for login Component", () => {
    const LoginComponent = render(<Login />);
    expect(LoginComponent).toMatchSnapshot();
  });
});

// import { fireEvent, render, screen } from "@testing-library/react";
// import Login from "./Login";

// describe("Validate the Login page", () => {
//   test("Input validation", () => {
//     render(<Login />);
//     const userInput = screen.getByRole("textbox");
//     fireEvent.change(userInput, { target: { value: "a" } });
//     expect(userInput.value).toBe("a");
//   });
// });

// describe("Click event testing with button", () => {
//   test("Button Onclick event test", () => {
//     render(<Login />);
//     const btn = screen.getByRole("button");
//     fireEvent.click(btn);
//     expect(screen.getByText("Update Button Clicked")).toBeInTheDocument();
//   });
// });
