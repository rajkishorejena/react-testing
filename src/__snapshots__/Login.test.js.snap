// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Snapshot testing for login Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; min-height: 100vh; display: flex; justify-items: center; align-items: center;"
      >
        <input
          name="username"
          placeholder="please enter your name"
          type="text"
          value=""
        />
        <label />
        <button>
          Update
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; min-height: 100vh; display: flex; justify-items: center; align-items: center;"
    >
      <input
        name="username"
        placeholder="please enter your name"
        type="text"
        value=""
      />
      <label />
      <button>
        Update
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
